components/widgets/menu/menu_items/useWords.tsx

```diff
@@ -42,22 +45,41 @@ export default function useWords(highestLimit: LimitSummary | false, isAdminUser
                 onClick={openPricingModal}
             >
                 {chunks}
-            </a>
-        ),
+            </a>),
+
     };
+
+    if (!isAdminUser && (usageRatio >= limitThresholds.danger || usageRatio >= limitThresholds.exceeded)) {
+        values.callToAction = intl.formatMessage({
+            id: 'workspace_limits.menu_limit.notify_admin',
+            defaultMessage: 'Notify admin',
+        });
+        values.a = (chunks: React.ReactNode | React.ReactNodeArray) => <NotifyAdminCTA ctaText={chunks}/>;
+    }
+
     switch (highestLimit.id) {
     case LimitTypes.messageHistory: {
         let id = t('workspace_limits.menu_limit.warn.messages_history');
         let defaultMessage = 'You’re getting closer to the free {limit} message limit. <a>{callToAction}</a>';
         values.limit = intl.formatNumber(highestLimit.limit);
         if (usageRatio >= limitThresholds.danger) {
-            id = t('workspace_limits.menu_limit.critical.messages_history');
-            defaultMessage = 'You’re close to hitting the free {limit} message history limit <a>{callToAction}</a>';
+            if (isAdminUser) {
+                id = t('workspace_limits.menu_limit.critical.messages_history');
+                defaultMessage = 'You’re close to hitting the free {limit} message history limit <a>{callToAction}</a>';
+            } else {
+                id = t('workspace_limits.menu_limit.critical.messages_history_non_admin');
+                defaultMessage = 'You\'re almost at the message limit. Your admin can upgrade your plan for unlimited messages. <a>{callToAction}</a>';
+            }
         }
         if (usageRatio >= limitThresholds.exceeded) {
-            id = t('workspace_limits.menu_limit.over.messages_history');
-            defaultMessage = 'You’re over the free message history limit. You can only view up to the last {limit} messages in your history. <a>{callToAction}</a>';
-            values.limit = inK(highestLimit.limit);
+            if (isAdminUser) {
+                id = t('workspace_limits.menu_limit.over.messages_history');
+                defaultMessage = 'You’re over the free message history limit. You can only view up to the last {limit} messages in your history. <a>{callToAction}</a>';
+                values.limit = inK(highestLimit.limit);
+            } else {
+                id = t('workspace_limits.menu_limit.over.messages_history_non_admin');
+                defaultMessage = 'You\'re over your message limit. Your admin can upgrade your plan for unlimited messages. <a>{callToAction}</a>';
```

Comment:
Do we also need to add this for all the other cases below besides `MessageHistory` limits?

Reply:
Oh ok - just to make sure I am following, the links to `View Plans`/ `Notify Admin` are being updated for all limits correct?